### PR TITLE
Added ability to copy/paste annotation guide with assets

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
Related #6536

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **Bug Fixes**
  - Resolved an issue where copying/pasting annotation guides with assets between different resources failed due to an error message.

- **Refactor**
  - Renamed `GuidePage` component to `AnnotationGuidePage` and refactored logic for fetching and saving guides.
  - Updated component structure and behavior in the annotation guide editor.

- **Style**
  - Adjusted the height property in the annotation guide editor for better layout consistency.

- **New Features**
  - Added a method to handle updating related assets in the annotation guides.

### CODE REVIEW

File: cvat/apps/engine/views.py [hunk 1]
```diff
@@ -2788,6 +2793,67 @@ class AnnotationGuidesViewSet(
     ordering = "-id"
     iam_organization_field = None
 
+    def _update_related_assets(self, request, guide: AnnotationGuide):
```
Comment (azhavoro) [inline cvat/apps/engine/views.py:2796]: There are many db objects being modified here, so I suggest wrapping this function as an atomic transaction

File: cvat/apps/engine/views.py [hunk 2]
```diff
@@ -2788,6 +2793,67 @@ class AnnotationGuidesViewSet(
...
+                    guide.markdown = guide.markdown.replace(
+                        f'(/api/assets/{asset_id})',
+                        f'(/api/assets/{copied_asset.uuid})',
+                    )
+
+                    markdown_assets.append(copied_asset)
+                else:
+                    markdown_assets.append(db_asset)
+
+            handled_assets[asset_id] = True
```
Comment (azhavoro) [inline cvat/apps/engine/views.py:2850]: Seems set can be used here

File: cvat/apps/engine/views.py [hunk 3]
```diff
@@ -2788,6 +2793,67 @@ class AnnotationGuidesViewSet(
...
+
+                    guide.markdown = guide.markdown.replace(
+                        f'(/api/assets/{asset_id})',
+                        f'(/api/assets/{copied_asset.uuid})',
+                    )
+
+                    os.makedirs(copied_asset.get_asset_dir())
+                    shutil.copyfile(
+                        os.path.join(os.path.join(db_asset.get_asset_dir(), db_asset.filename)),
+                        os.path.join(os.path.join(copied_asset.get_asset_dir(), db_asset.filename)),
```
Comment (azhavoro) [inline cvat/apps/engine/views.py:2838]:
```suggestion
                        os.path.join(db_asset.get_asset_dir(), db_asset.filename),
                        os.path.join(copied_asset.get_asset_dir(), db_asset.filename),
```

File: cvat/apps/engine/views.py [hunk 4]
```diff
@@ -2788,6 +2793,67 @@ class AnnotationGuidesViewSet(
...
+                    )
+
+                    guide.markdown = guide.markdown.replace(
+                        f'(/api/assets/{asset_id})',
+                        f'(/api/assets/{copied_asset.uuid})',
+                    )
```
Comment (azhavoro) [inline cvat/apps/engine/views.py:2844]: It's already been done in the line 2830-2833

File: cvat/apps/engine/views.py (hunk 1 above)
Comment (bsekachev) [reply cvat/apps/engine/views.py:2922]: I added transaction.atomic where we copy existing assets

**Review by azhavoro — APPROVED** [review]

**Pull request merged**